guides/2022-04-05-build-a-conduit-connector.mdx

```diff
@@ -0,0 +1,318 @@
+3. Write - Write record to datastore.
+4. Teardown - Clean up and close connection.
+
+Learn more about the connector lifecycle in [The Guide to the Conduit Connector Lifecyle](/guides/guide-to-conduit-connector-lifecycle).
+
+Connectors can be created in many programming languages because Conduit Plugins leverage GRPC. For this example, we are going to use the Golang SDK. 
+
+Also, for this example, the datastore we are building the connector for is Algolia. Algola is TODO. As a destination, this connector allows for real-time syncing between any Conduit Source and Algolia.
+
+To communicate with Algolia, we can use the [Algolia Golang library](https://www.algolia.com/doc/api-client/getting-started/install/go/?client=go).
+
+### Connector File Structure
+
+Next, let's take a look at an example connector file structure:
+
+- `destination.go` - The main destination code.
+- `object.go` - Helper functions to serialize records into JSON for Algolia
+- `spec.go` - The specification defines connector metadata and parameters.
+- `cmd/algolia/main.go` - Entry point
+
+## Setup
+
+Let's bootstrap our new connector. We are going to name our connector `conduit-connector-algolia`. It's best practice to use the `conduit-connector-NAME` format so the community can find connectors. 
+
+1. Create a new folder for your connector:
+
+```sh
+mkdir -p /where/you/want/your/connector
+```
+
+2. Init Go Modules:
+
+```sh
+go mod init
+```
+
+3. Create the `main.go` entrypoint.
+
+```
+mkdir cmd/algolia && touch cmd/algolia/main.go
+```
+4. Create 
+
+```
+touch destination.go spec.go
+```
+
+5. Install Go dependencies
+
+```
+go get
+```
+
+## Specification
+
+First, let's build `spec.go`. The Specification
+
+```go
+package algolia
+
+import sdk "github.com/conduitio/conduit-connector-sdk"
+
+func Specification() sdk.Specification {
+	return sdk.Specification{
+		Name:        "algolia",
+		Summary:     "A destination connector for Algolia",
+		Description: "TBD",
+		Version:     "v0.1.0",
+		Author:      "Meroxa, Inc.",
+		DestinationParams: map[string]sdk.Parameter{
+			DestinationConfigAPIKey: {
+				Required:    true,
+				Description: "The API key for Algolia.",
+			},
+			DestinationConfigApplicationID: {
+				Required:    true,
+				Description: "The Application ID for Algolia.",
+			},
+			DestinationConfigIndexName: {
+				Required:    true,
+				Description: "The Algolia index where records get written into.",
+			},
+		},
+		SourceParams: nil,
+	}
+}
+```
+
+## Entrypoint
+
+Next, let's look at `cmd/algolia/main.go`. This is our entrypoint. 
+
+```go
+package main
+
+import (
+	algolia "github.com/conduitio/conduit-connector-algolia"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+func main() {
+	sdk.Serve(
+		algolia.Specification,
+		nil,
+		algolia.NewDestination,
+		)
+}
+```
+
+`sdk.Serve` is TODO. Notice `nil` as the second parameter. This is due to this connector being just a Destination.
```

Comment:
"`sdk.Serve` starts the plugin and takes care of its whole lifecycle" - did you mean something like this?